src/Features/Core/Portable/CommentSelection/CommentSelectionInfo.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+namespace Microsoft.CodeAnalysis.CommentSelection
+{
+    internal struct CommentSelectionInfo
+    {
+        public CommentSelectionInfo(bool supportsSingleLineComment, bool supportsBlockComment, string singleLineCommentString, string blockCommentStartString, string blockCommentEndString) : this()
+        {
+            SupportsSingleLineComment = supportsSingleLineComment;
+            SupportsBlockComment = supportsBlockComment;
+            SingleLineCommentString = singleLineCommentString;
+            BlockCommentStartString = blockCommentStartString;
+            BlockCommentEndString = blockCommentEndString;
+        }
+
+        public bool SupportsSingleLineComment { get; }
+        public bool SupportsBlockComment { get; }
+
+        public string SingleLineCommentString { get; }
+
+        public string BlockCommentStartString { get; }
+        public string BlockCommentEndString { get; }
+    }
```

Comment:
Ryan, this type is essentially what you return.  Note that you have the flexibility of saying if you support either/both/none of Single/Block comments.  We query for this object below and we pass you a SourceText and a text span.  Note: if you need a Document, let me know and i can pass that instead.